File: Event/src/main/java/com/esprit/event/DAO/entities/Event.java
package com.esprit.event.DAO.entities;

import jakarta.persistence.*;
import lombok.*;
import org.attoparser.dom.Text;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;


@Entity
@Getter
@Setter
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idEvent;
    private String eventName;

    private String eventDescription;
    private LocalDateTime eventDate;
    @Enumerated(EnumType.STRING)
    private Category eventCategory;
    private String imageUrl;
    @ManyToOne
    private Centre centre;

    // Proper mapping for participant IDs
    @ElementCollection
    @CollectionTable(
            name = "event_participants",
            joinColumns = @JoinColumn(name = "event_id")
    )
    @Column(name = "user_id")  // This matches the column in your join table
    private List<Integer> participants = new ArrayList<>();  // Initialize to avoid NPE
    private Integer eventCreator;
}

File: full-version/src/app/models/Event.ts
import { CategoryEnum } from './CategoryEnum';
import { Centre } from './Centre';
import { User } from './User';

export class Event {
  idEvent: number;
  eventName: string;
  eventDescription: string;
  eventDate: string; // Use string because Angular handles date conversion
  eventCategory: CategoryEnum;
  imageUrl:  string;
  centre: Centre;
  participants: User[];
  eventCreator: User;
  eventDateOnly?: string;
  eventTimeOnly?: string;
}
